app/views/tasks/index.html.erb

```diff
@@ -0,0 +1,28 @@
+
+
+<!-- <h1>TaskList</h1>
+<p>Keeping Track of To-do's</p> -->
+
+<ul>
+  <% @tasks.each do |task| %>
+  <li>
+
+    <input type="checkbox" <%="checked" if task.completed? %> disabled >
+
+    <strong> <%= link_to task.name, task_path(task.id)%></strong> : <%= task.description %> <%= task.completion_date %>
+
+    <% if task.completed? %>
+      <%= button_to("Unmark Complete", uncompleted_tasks_path(task.id), method: :patch) %>
+    <% else %>
+      <%= button_to("Complete this Task", completed_tasks_path(task.id), method: :patch, data: {confirm: "Are you sure you want to mark your task as 'Complete'?"}) %>
+    <% end %>
+
+    <%= button_to("Delete this Task", task_path(task.id), method: :delete, data: {confirm: "Are you sure you want to Delete the Task?"}) %>
+
+    <%= button_to("Edit this Task", edit_task_path(task.id)) %>
+
```

Comment:
On line 22, you should be using `link_to` instead of `button_to`. `button_to` creates a `<button>` element, and the default HTTP verb for a button is `POST`. That means that clicking on this link results in a routing error, since the only route for the `edit_task_path` is a `GET`.